toolchain/sem_ir/formatter.cpp

```diff
@@ -924,6 +918,31 @@ class Formatter {
     out_ << "\n";
   }
 
+  auto FormatPendingConstantValue(bool space_before) -> void {
+    if (!pending_constant_value_.is_valid() ||
+        pending_constant_value_.is_constant()) {
```

Comment:
I'll note the irony here of changing an invalid value to not constant, because the assignment is set up to run unconditionally. That makes the `is_constant` consistently run on the second pass, even for initially invalid values. Had you considered an early return before assigning, dropping the indent? :)

Reply:
Changed to early return, added comment to explain the use of `NotConstant` as a sentinel, and simplified the condition to a direct comparison against `NotConstant` to make it more obvious that's the value we're excluding here.